week-6/Homework/mandatory/1-alarmclock/alarmclock.js

```diff
@@ -1,4 +1,71 @@
-function setAlarm() {}
+function setAlarm() {
+  //get the value of the input field
+  let inputField = document.querySelector("#alarmSet").value;
+
+  //set the title to the correct value
+  const timeRemaining = document.querySelector("#timeRemaining");
+  timeRemaining.textContent = `Time Remaining: ${inputField}:00`;
+
+  let myAlarm;
+  ///
+  document.getElementById("set").hidden = true;
+
+  // add `pause` functionality
+  document.getElementById("set").textContent = "Pause";
+
+  //create new button Pause
+  let pauseButton = document.createElement("button");
+  pauseButton.textContent = "Pause";
+  let divButton = document.querySelector(".centre >:last-child");
+  divButton.prepend(pauseButton);
+  pauseButton.className = "pauseButton";
+
+  document.querySelector(".pauseButton").addEventListener("click", () => {
+    clearInterval(myAlarm);
+    document.querySelector(".pauseButton").hidden = true;
+
+    let restartButton = document.createElement("button");
+    restartButton.textContent = "Restart";
+    let divButton = document.querySelector(".centre >:last-child");
+    divButton.prepend(restartButton);
+    restartButton.className = "restartButton";
+
+    //restart functionality
+    document.querySelector(".restartButton").addEventListener("click", () => {
```

Comment:
What about if I want to unpause the timer? The restart button just seems to reset it. Think about how you could maybe control the flow of execution in the timer in such a way as you could easily toggle the paused state (something like `if (paused)`).

Reply:
Thanks for your feedback. I have added the ability to unpause it. Thanks for your assistance.